guide/interactions/slash-command-permissions.md

```diff
@@ -0,0 +1,137 @@
+
+```js
+client.on('messageCreate', async message => {
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const command = await client.guilds.cache.get('123456789012345678')?.commands.fetch('876543210987654321');
+
+		const permissions = [
+			{
+				id: '224617799434108928',
+				type: 'USER',
+				permission: false,
+			},
+		];
+
+		await command.permissions.add({ permissions });
+	}
+});
+```
+
+Now you have successfully denied the user who's `id` you used access to this application command.
+If you have a command that is disabled by default and you want to grant someone access to use it, do as follows:
+
+```js {11}
+client.on('messageCreate', async message => {
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const command = await client.guilds.cache.get('123456789012345678')?.commands.fetch('876543210987654321');
+
+		const permissions = [
+			{
+				id: '224617799434108928',
+				type: 'USER',
+				permission: true,
+			},
+		];
+
+		await command.setPermissions({ permissions });
+	}
+});
+```
+
+And that's how you use slash command permissions to deny or allow access to a single user!
+
+## Role permissions
+
+Now you may want to allow (or deny) multiple users the usage of a command. In this scenario you can apply a permission that is scoped to a role instead of a user:
+
+```js {10,11}
+client.on('messageCreate', async message => {
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const command = await client.guilds.cache.get('123456789012345678')?.commands.fetch('876543210987654321');
+
+		const permissions = [
+			{
+				id: '464464090157416448',
+				type: 'ROLE',
+				permission: false,
+			},
+		];
+
+		await command.permissions.add({ permissions });
+	}
+});
+```
+
+With this, you have successfully denied an entire role access to this command. If you want to allow a certain role to access this command, you repeat this procedure while setting `permission: true` in the `ApplicationCommandPermissionData` as shown below:
+
+```js {11}
+client.on('messageCreate', async message => {
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const command = await client.guilds.cache.get('123456789012345678')?.commands.fetch('876543210987654321');
+
+		const permissions = [
+			{
+				id: '464464090157416448',
+				type: 'ROLE',
+				permission: true,
+			},
+		];
+
+		await command.permissions.add({ permissions });
+	}
+});
```

Comment:
```suggestion
Permissions may also be denied (or allowed) at a role scope instead of a single user:

<!-- eslint-skip -->

```js {4-5}
const permissions = [
	{
		id: '464464090157416448',
		type: 'ROLE',
		permission: false,
	},
];

await command.permissions.add({ permissions });
```